src/Emitter.ts

```diff
@@ -19,91 +19,298 @@ export enum Events {
    */
   namedExpressionRemoved: (namedExpressionName: string, changes: ExportedChange[]) => any,
 
   /**
-   * The event is triggered upon changing values in a specified location.
+   * Occurs when values in a specified location are changed and cause recalculation.
    * 
    * @event
    * 
    * @param {ExportedChange[]} changes the values and location of applied changes
    *
+   * @example
+   * ```js
+   * const hfInstance = HyperFormula.buildFromArray([
+   * ['1', '2', '=A1'],
+   * ]);
+   *
+   * // define a function to be called when the event occurs
+   * const handler = ( ) => { console.log('baz') }
+   * 
+   * // subscribe to the 'valuesUpdated' event, pass the handler
+   * hfInstance.on('valuesUpdated', handler);
+   * 
+   * // trigger recalculation, for example, with the 'setCellContents' method
+   * // the console prints 'baz' each time a value change triggers recalculation
+   * const changes = hfInstance.setCellContents({ col: 3, row: 0, sheet: 0 }, [['=B1']]);
+   *
+   * // unsubscribe from the 'valuesUpdated' event
+   * hfInstance.off('valuesUpdated', handler);
+   *
+   * // trigger another recalculation
+   * // this time, the console doesn't print anything
+   * const changes = hfInstance.setCellContents({ col: 3, row: 0, sheet: 0 }, [['=A1']]);
+   * ```
+   * 
    * @category Values
    */
   valuesUpdated: (changes: ExportedChange[]) => any,
 
   /**
-   * The event is triggered upon suspension of evaluation.
+   * Occurs when evaluation is suspended.
    *
    * @event
+   * 
+   * @example
+   * ```js
+   * const hfInstance = HyperFormula.buildFromSheets({
+   * MySheet1: [ ['1'] ],
+   * MySheet2: [ ['10'] ]
+   * });
+   *
+   * // define a function to be called when the event occurs
+   * const handler = ( ) => { console.log('baz') }
+   * 
+   * // subscribe to the 'evaluationSuspended' event, pass the handler
+   * hfInstance.on('evaluationSuspended', handler);
+   * 
+   * // suspend evaluation
+   * // the console prints 'baz' each time evaluation is suspended
+   * hfInstance.suspendEvaluation();
+   *
+   * // resume evaluation
+   * hfInstance.resumeEvaluation();
+   * 
+   * // unsubscribe from the 'evaluationSuspended' event
+   * hfInstance.off('evaluationSuspended', handler);
+   *
+   * // suspend evaluation again
+   * // this time, the console doesn't print anything
+   * hfInstance.suspendEvaluation();;
+   * ```
    *
    * @category Batch
    */
   evaluationSuspended: () => any,
 
   /**
-   * The event is triggered upon resuming of evaluation.
+   * Occurs when evaluation is resumed.
    *
    * @event
    *
    * @param {ExportedChange[]} changes the values and location of applied changes
    *
+   * @example
+   * ```js
+   * const hfInstance = HyperFormula.buildFromSheets({
+   * MySheet1: [ ['1'] ],
+   * MySheet2: [ ['10'] ]
+   * });
+   *
+   * // define a function to be called when the event occurs
+   * const handler = ( ) => { console.log('baz') }
```

Comment:
We should show that this handler have an argument

```suggestion
   * // define a function to be called when the event occurs
   * const handler = ( changes ) => { console.log('baz') }
```